src/asl.ts

```diff
@@ -428,21 +428,52 @@ export class ASL {
           return hoist && self.doHoist(node)
             ? hoist(updatedChild)
             : updatedChild;
+        } else if (isBinaryExpr(node)) {
+          const updated = visitEachChild(node, (expr) =>
+            normalizeAST(expr, hoist)
+          );
+          return hoist && self.doHoist(node) ? hoist(updated) : updated;
+        } else if (
+          isArrayLiteralExpr(node) &&
+          !node.items.some((i) => isFunctionExpr(i))
+        ) {
+          // if the array contains functions, just leave it, it is probably for Parallel.
+          // This is short sighted, but should work now
+          const updated = visitEachChild(node, (expr) =>
+            normalizeAST(expr, hoist)
+          );
+
+          return hoist && self.doHoist(node) ? hoist(updated) : updated;
         }
         return visitEachChild(node, (expr) => normalizeAST(expr, hoist));
       }
     );
 
-    const states = this.execute(this.decl.body);
+    const states = this.evalStmt(this.decl.body);
 
     const start = this.transition(this.decl.body);
     if (start === undefined) {
       throw new Error("State Machine has no States");
     }
 
+    /**
+     * Add some hard to manufacture constants to the machine.
+     *
+     * TODO: only add this when used?
```

Comment:
I think this is covered by https://github.com/functionless/functionless/pull/316

Reply:
Yep